modules/core/src/test/java/org/apache/ignite/internal/binary/BinaryFieldExtractionSelfTest.java

```diff
@@ -193,6 +193,47 @@ public void testChangeTypeIdOfBinaryFieldCaseNotFoundExpectedTypeId() throws Exc
         assertThrows(log, () -> timeBinField.value(timeValBinObj), BinaryObjectException.class, expMsg);
     }
 
+    /**
+     * Checking the exception and its text when changing the typeId by spoil
+     * internal array of a BinaryField for case of not finding the actual
+     * BinaryType.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testChangeTypeIdOfBinaryFieldCaseNotFoundActualTypeId() throws Exception {
+        BinaryMarshaller marsh = createMarshaller();
+
+        TimeValue timeVal = new TimeValue(11111L);
+
+        BinaryObjectImpl timeValBinObj = toBinary(timeVal, marsh);
+
+        BinaryFieldEx timeBinField = (BinaryFieldEx)timeValBinObj.type().field("time");
+
+        int beforeTypeId = timeValBinObj.typeId();
+
+        String fieldType = binaryContext(marsh).metadata(timeValBinObj.typeId()).fieldTypeName(timeBinField.name());
+
+        Field startField = U.findField(timeValBinObj.getClass(), "start");
+        int start = (int)startField.get(timeValBinObj);
+
+        Field arrField = U.findField(timeValBinObj.getClass(), "arr");
+        byte[] arr = (byte[])arrField.get(timeValBinObj);
+        arr[start + 4] = 42;
```

Comment:
public static final int TYPE_ID_POS = 4; looks like public, if u mention it here.

Reply:
looks like more accurate would be code like : arr[start + 4] = beforeTypeId + 1; for example.